app/models/test_track/visitor.rb

```diff
@@ -62,10 +62,6 @@ def assignment_json
     end
   end
 
-  def split_registry
-    @split_registry ||= TestTrack::Remote::SplitRegistry.to_hash
```

Comment:
should we keep this and its usages above? it makes the vary code deterministic because its not dependent on the split registry being consistent and in the cache or available from the remote server.

i'm thinking about the (admittedly unlikely) case where the cache gets invalidated while running the vary algo, and either the weights change or the split registry goes offline.

Reply:
yeah, i mean, at scale unlikely happens a lot. problem is that this is a hash, and that's directly at odds with the object envy issue without a major refactor to be passing around a `SplitRegistry` instance of some sort that would implement a bunch of new methods instead of keeping this hash ref. So I see three paths:

* Accept this race condition - kinda ugly semantically
* revert, maybe keep the caching of nil which I like and eat the object envy for now
* Revert and go once more into the breach and take everything that used to hold reference to a registry hash to hold an instance of a nice object.

I'm sort of angling for option 2.